fix(github-actions): checkout repo before running inactivity bot

Akshat8510 opened the pull request:
**Description**  
This PR fixes the workflow failure in `bot-inactivity-unassign-phase1` by ensuring the repository is checked out before executing the inactivity bot script.

Previously, GitHub Actions could not locate the script:

``bash
bash: .github/scripts/inactivity_unassign_phase1.sh: No such file or directory
``

This happened because the workflow did **not** include a checkout step.  
Adding `actions/checkout@v4` resolves the issue.

---

**Changes**
- Added checkout step before running the inactivity unassign script.

---

**Related issue(s)**
Fixes #964

---

**Notes for reviewer**
- Verified on fork: workflow now finds the script correctly.
- Expected to run successfully on the next scheduled execution.

---

**Checklist**
- [x] Documented (N/A – workflow tweak)
- [x] Tested locally / on fork
- [x] Fix validated in logs

exploreriii commented on .github/workflows/bot-inactivity-unassign-phase1.yml:
```diff
@@ -15,6 +15,8 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
+      - name: Checkout repository
```
Hello, great the full commit hash is
8e8c483db84b4bee98b60c0593521ed34d9990e8
https://github.com/actions/checkout/commit/8e8c483db84b4bee98b60c0593521ed34d9990e8

<img width="1235" height="223" alt="Image" src="https://github.com/user-attachments/assets/e19655bd-9a40-4716-b536-3b20f799c02b" />

Akshat8510 commented:
Hi @exploreriii,
I’ve made all the updates you mentioned:

Updated the checkout action to use the full commit hash
Adjusted the cron time to 12:00 UTC
Added the corresponding entry under Fixed in the changelog

Please take your time reviewing it — no rush at all.
If you spot anything else or feel something should be improved, I’m happy to update it!
